Subject:      Allow for TBB_LIBRARY_DIR to specify custom lib directory in a multiple TBB installation
Path Author:  Orson Teodoro <orsonteodoro@hotmail.com>
Date:         Tue Jun 29 01:19:44 AM PDT 2021 (Unix timestamp: 1624954784)

Copyright notice for common/cmake/FindTBB.cmake:

## Copyright 2009-2021 Intel Corporation
## SPDX-License-Identifier: Apache-2.0

diff -urp embree-3.13.0.orig/common/cmake/FindTBB.cmake embree-3.13.0/common/cmake/FindTBB.cmake
--- embree-3.13.0.orig/common/cmake/FindTBB.cmake	2021-06-29 01:39:13.154242218 -0700
+++ embree-3.13.0/common/cmake/FindTBB.cmake	2021-06-29 01:41:44.724110776 -0700
@@ -22,6 +22,8 @@
 # The user may specify the following variables to help the search process:
 # - TBB_ROOT
 # - TBB_INCLUDE_DIR
+# - TBB_LIBRARY_DIR									# oteodoro - added line
+# - TBB_SOVER										# oteodoro - added line
 #
 # After the script has run successfully, there is a target TBB, as well as
 # component targets TBB::<COMPONENT>, e.g. TBB::tbbmalloc.
@@ -37,10 +39,15 @@
 #===============================================================================
 
 # oteodoro - debug message to show preference between legacy TBB and oneTBB
+# oteodoro - add alternative library path in Linux
 
 # We use INTERFACE libraries, which are only supported in 3.x
 cmake_minimum_required(VERSION 3.1)
 
+if (TBB_LIBRARY_DIR)									# oteodoro added line
+  set(FIND_PATH_OPTS_ NO_DEFAULT_PATH)							# oteodoro added line
+endif()											# oteodoro added line
+
 # These two are used to automatically find the root and include directories.
 set(_TBB_INCLUDE_SUBDIR "include")
 set(_TBB_HEADER "tbb/tbb.h")
@@ -197,6 +204,7 @@ macro(rk_tbb_reuse_existing_target_compo
     endif()
 
     find_path(TBB_INCLUDE_DIR
+      ${FIND_PATH_OPTS_}												# oteodoro - added line
       NAMES "${_TBB_HEADER}"
       PATHS "${TBB_INCLUDE_DIRS}")
 
@@ -272,6 +280,7 @@ endfunction()
 
 function(rk_tbb_find_include_directory)
   find_path(TBB_INCLUDE_DIR
+    ${FIND_PATH_OPTS_}										# oteodoro - added line
     NAMES "${_TBB_HEADER}"
     HINTS "${TBB_ROOT}/${_TBB_INCLUDE_SUBDIR}"
     NO_PACKAGE_ROOT_PATH)
@@ -346,6 +355,7 @@ function(rk_tbb_find_library COMPONENT_N
     file(GLOB LIB_PATHS PATHS ${TBB_ROOT}/lib/intel64/gcc*)
     list(REVERSE LIB_PATHS)
     list(APPEND LIB_PATHS
+      ${TBB_LIBRARY_DIR}										# oteodoro - added line
       ${TBB_ROOT}/lib
       ${TBB_ROOT}/lib/x86_64-linux-gnu
       ${TBB_ROOT}/lib64
@@ -356,7 +366,7 @@ function(rk_tbb_find_library COMPONENT_N
   # variable will not point to a symlink. This makes installing TBB as a
   # dependency easier.
   if (UNIX)
-    set(LIB_NAME lib${LIB_NAME}.so.2 ${LIB_NAME})
+    set(LIB_NAME lib${LIB_NAME}.so.${TBB_SOVER} lib${LIB_NAME}.so.2 ${LIB_NAME})			# oteodoro - added a particular sover to match
   endif()
 
   find_library(${LIB_VAR}
Only in embree-3.13.0/common/cmake: FindTBB.cmake.orig
Only in embree-3.13.0/common/cmake: FindTBB.cmake.rej
